src/utils/hdKey.ts

```diff
@@ -1,79 +1,104 @@
 // Copyright © Aptos Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-import nacl from "tweetnacl";
 import { hmac } from "@noble/hashes/hmac";
 import { sha512 } from "@noble/hashes/sha512";
-import { hexToBytes } from "@noble/hashes/utils";
+import * as bip39 from "@scure/bip39";
 
-export type Keys = {
+export type DerivedKeys = {
   key: Uint8Array;
   chainCode: Uint8Array;
 };
 
-const pathRegex = /^m(\/[0-9]+')+$/;
+/**
+ * Aptos derive path is 637
+ *
+ * See https://github.com/satoshilabs/slips/blob/master/slip-0044.md
+ */
+export const APTOS_PATH_REGEX = /^m\/44'\/637'\/[0-9]+'\/[0-9]+'\/[0-9]+'?$/;
 
-const replaceDerive = (val: string): string => val.replace("'", "");
-
-const HMAC_KEY = "ed25519 seed";
+export const ED25519_KEY = "ed25519 seed";
```

Comment:
do we also need support for Secp256k1

Reply:
We may but, I'll have to do more research into what the output value needs to be from it.